cmd/cluster/cleanup_leaked_ec2.go

```diff
@@ -124,23 +124,14 @@ func (c *cleanup) Run(ctx context.Context) error {
 }
 
 func (c *cleanup) RemediateOCPBUGS23174(ctx context.Context) error {
-	awsmachines := &capav1beta2.AWSMachineList{}
-	if err := c.client.List(ctx, awsmachines, client.MatchingLabels{
-		"cluster.x-k8s.io/cluster-name": c.cluster.ID(),
-	}); err != nil {
-		return err
-	}
-
-	expectedInstances := map[string]bool{}
-	for _, awsmachine := range awsmachines.Items {
-		if awsmachine.Spec.InstanceID != nil {
-			expectedInstances[*awsmachine.Spec.InstanceID] = true
-		}
-	}
-	log.Printf("expected instances: %v", expectedInstances)
-
 	resp, err := c.awsClient.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
 		Filters: []types.Filter{
+			{
+				// We don't want to terminate pending because they would just be started and might not have been picked up by the MC yet.
+				// Importantly we also don't want to count terminated instances as leaked, as it causes confusion.
+				Name:   aws.String("instance-state-name"),
+				Values: []string{"running", "stopped"},
```

Comment:
Optional nit: there are constants available for these states if you would like to use them https://github.com/aws/aws-sdk-go-v2/blob/6dbfda7e9981fc0f5f03e6001526c7b8b5938257/service/ec2/types/enums.go#L2916-L2924

Reply:
types.InstanceStateName does not seem to be designed to be used for types.Filter :(
When doing so I have:
cannot use []types.InstanceStateName{…} (value of type []"github.com/aws/aws-sdk-go-v2/service/ec2/types".InstanceStateName) as []string value in struct literal

Reply:
It would still be `[]string` as it is a type alias for `string`

Reply:
Well, golang is not as much permissive as this, and it returns an error, because it is expecting an string and not a InstanceStateName :(

Reply:
```suggestion
				Values: []string{string(types.InstanceStateNameRunning), string(types.InstanceStateNameStopped)},
```

Reply:
Ok, indeed, forcing the string type, it works... Not sure it is very "elegant" but if you prefer ;)